vault/provider.go

```diff
@@ -761,6 +761,14 @@ var (
 			Resource:      UpdateSchemaResource(secretsSyncAssociationResource()),
 			PathInventory: []string{"/sys/sync/destinations/{type}/{name}/associations/set"},
 		},
+		"vault_plugin": {
+			Resource:      UpdateSchemaResource(pluginResource()),
+			PathInventory: []string{"/sys/plugins/catalog/{type}/{name}"},
+		},
+		"vault_plugin_pinned_version": {
```

Comment:
@vinay-gopalan WDYT about creating a new resource for this set of API endpoints? One alternative I was considering was whether I should simply add a `pinned_version` boolean into the `vault_plugin` resource. Pins can't be created without the plugin first existing, and plugins can't be deleted if they're currently pinned, so I thought it would a) Make the dependencies between them easier to manage and b) make the terraform config a little bit simpler/less repetitive.

Reply:
That's definitely a great way to think about this. In general, if it is possible to manage the lifecycle of related Vault objects under a single TF resource, I would encourage it. In this case, since the separate `pinned_version` resource doesn't have any unique fields and uses the same fields as the plugin resource, this might be a good candidate to merge into one;

 A few thoughts I had while pondering the single-resource workflow in this case to be wary of:
- It will be possible to add a `pinned_version` bool. If the user sets this value, we will pin a provided version using the new plugin pin API after we register the plugin. In this scenario, if a `pinned_version` is provided, we will force the `version` parameter to be `Required` instead of `Optional`
- The lifecycle of the registered plugin itself can be managed via TF easily — CRUD should be seamless w/ the single-resource case
- The lifecycle of the pinned version in the single-resource case will be slightly more involved/complicated — for example, if we want to delete a pinned version for a plugin using this single-resource, we might have to write a `CustomizeDiff` func that will track if the `pinned_version` bool was in fact changed from `true` -> `false` (we may not want to call the DELETE endpoint for a pinned version if the `pinned_version` bool was never actually set to `true`, not sure if Vault returns an error in this case for "pinned version not found"). This is one caveat that might lead us to keep the resources separate if we find that to be simpler. 
- Also, if we envision this ecosystem to evolve, and we foresee cases where having a separate plugin resource that is modular and only does one thing is more beneficial, I think we should go that route now as opposed to later.

There might be some concepts/roadmap items about registering plugins that I might not be aware of and may need to get up to speed on, happy to chat more on this to figure out what is the best UX for our users when using this feature through TF!

Reply:
Thanks for the thoughts, much appreciated! I'm leaning towards separate resources because of the complexity for pinned version lifecycle that you mentioned. As another example, if the diff removes the pinned version from one `vault_plugin` and adds it to another of the same name and type, we'll want to do a delete for the first resource's pinned version and a create for the other's. As such, it's important that we order those correctly so we don't delete the freshly created pin, but AFAIU I don't think it's easy to tell Terraform about that ordering. I think it's also possible the pinned version will evolve in future to include semver constraints, and as you say it feels easier to support those changes with a separate resource.

Reply:
That makes sense, thanks for the additional context!